src/libcrun/linux.c

```diff
@@ -3443,6 +3508,13 @@ libcrun_run_linux_container (libcrun_container_t *container, container_entrypoin
       if (UNLIKELY (ret < 0))
         return crun_make_error (err, errno, "close");
 
+      /* any systemd notify socket open_tree FD is pointless to keep around in the parent */
+      if (get_private_data (container)->notify_socket_tree_fd >= 0)
+        {
+          close (get_private_data (container)->notify_socket_tree_fd);
+          get_private_data (container)->notify_socket_tree_fd = -1;
```

Comment:
you can simplify these two lines with: `close_and_reset (&(get_private_data (container)->notify_socket_tree_fd));